miden-crypto/src/dsa/ecdsa_k256/mod.rs

```diff
@@ -0,0 +1,311 @@
+//! ECDSA (Elliptic Curve Digital Signature Algorithm) signature implementation over secp256k1
+//! curve.
+
+use alloc::{string::ToString, vec::Vec};
+
+use k256::{
+    ecdh::diffie_hellman,
+    ecdsa::{RecoveryId, SigningKey, VerifyingKey, signature::hazmat::PrehashVerifier},
+};
+use rand::{CryptoRng, RngCore};
+use thiserror::Error;
+
+use crate::{
+    Felt, SequentialCommit, StarkField, Word,
+    ecdh::{EphemeralPublicKey, SharedSecret},
+    utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable},
+};
+
+#[cfg(test)]
+mod tests;
+
+// CONSTANTS
+// ================================================================================================
+
+/// Length of secret key in bytes
+const SECRET_KEY_BYTES: usize = 32;
+/// Length of public key in bytes when using compressed format encoding
+pub(crate) const PUBLIC_KEY_BYTES: usize = 33;
+/// Length of signature in bytes using our custom serialization
+const SIGNATURE_BYTES: usize = 66;
+/// Length of signature in bytes using standard serialization i.e., `SEC1`
+const SIGNATURE_STANDARD_BYTES: usize = 64;
+/// Length of scalars for the `secp256k1` curve
+const SCALARS_SIZE_BYTES: usize = 32;
+
+// SECRET KEY
+// ================================================================================================
+
+/// Secret key for ECDSA signature verification over secp256k1 curve.
+pub struct SecretKey {
+    inner: SigningKey,
+}
+
+impl SecretKey {
+    /// Generates a new random secret key using the OS random number generator.
+    #[cfg(feature = "std")]
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Self {
+        let mut rng = rand::rng();
+
+        Self::with_rng(&mut rng)
+    }
+
+    /// Generates a new secret key using the provided random number generator.
+    pub fn with_rng<R: CryptoRng + RngCore>(rng: &mut R) -> Self {
+        use k256::elliptic_curve::rand_core::SeedableRng;
+
+        let mut seed = [0_u8; 32];
+        rand::RngCore::fill_bytes(rng, &mut seed);
+
+        let mut rng = rand_hc::Hc128Rng::from_seed(seed);
```

Comment:
I think this approach works, and I think this is exactly what I suggested in https://github.com/0xMiden/crypto/pull/484#discussion_r2312083984.

Would be good to add a comment explaining why we do this here, and we should also create an issue to come back to this once new versions of Rust crypto crates are available.